src/shared/dissect-image.c

```diff
@@ -2224,13 +2224,17 @@ int dissected_image_mount(
         if (r < 0)
                 return r;
 
-        int slash_boot_is_available = 0;
-        if (where) {
-                r = slash_boot_is_available = mount_point_is_available(where, "/boot", /* missing_ok = */ true);
-                if (r < 0)
-                        return r;
-        }
-        if (!where || slash_boot_is_available) {
+        bool slash_boot_is_available = true;
```

Comment:
so i am not convinced this really makes sense.

/boot/efi is just dumb. it's OK to *look* there. It's not ok to actually automatically mount things there i am very sure.

Sorry, but let's push distros to stop doing this nonsense, let's not just follow suit. it makes really no sense to nest these dirs.

Reply:
So you prefer having `systemd-dissect` failing with "sorry but refusing mounting ESP on top of /boot/efi" ? Or maybe just emitting the warning and mount the rest of the partitions ?

This makes little sense too considering that /boot/efi is common and is compliant with DPS. 

Omitting the support of /boot/efi in gpt-auto is fine and should indicate that /boot/efi is obsolete. But the current behavior of dissect is incorrect.